test/server/overload_manager_impl_test.cc

```diff
@@ -130,6 +130,26 @@ class OverloadManagerImplTest : public testing::Test {
     )EOF";
   }
 
+  std::string getConfigSimple() {
```

Comment:
Sorry what does this change have to do with Windows?

Reply:
the issue this change solves isnt technically isnt a windows specific change but the problem only occurs on windows right now

https://github.com/envoyproxy/envoy/pull/11335/commits/99d9187ade319554b1ec2d31581a0908b5b85601 has a little bit of the context

when there are two resources/triggers we're keeping track of [this line](https://github.com/envoyproxy/envoy/pull/11335/commits/99d9187ade319554b1ec2d31581a0908b5b85601#diff-69e6b63d0208ce9950f961fd3fa76af5L243) captures the second posted callback per ticker; It just so happens that the glibc++ and stdlib++ hashing in std::unordered_map vary from the msvc implementation so the second captured callback on windows is for the wrong resource and the test fails, on linux etc. the second captured callback is for the correct resource so the test just works

Reply:
cc @eziskind can you review this part of the change? Another option is to switch to absl::flast_hash_map which should be consistent on all platforms, though this sounds like a test that is already making strange assumptions.

Reply:
@mattklein123 why are you trying to write a differently broken test? 

The underlying issue is the two triggers on one test and relying on a hash for ordering is absurd. You can show this up by simply toggling fake_resource1 to 2 on line 248. I'd suggest accepting the patch and planning a rewrite to dropping the extra single-scope config later for correcting the test to cope with two triggers. But whatever.

Reply:
> @mattklein123 why are you trying to write a differently broken test?

I'm not. We should fix the test. However, I don't have the time right now to invest in reviewing this change to see if we are losing any test coverage. @eziskind wrote this code and should be able to take a look.

Reply:
This change looks good. Thanks @sunjayBhatia and @wrowe for debugging this issue!